lib/views/components/recording/recording_content.dart

```diff
@@ -55,56 +57,42 @@ class _RecordingPageState extends State<RecordingPage> {
-    speechProvider = Provider.of<SpeechToTextProvider>(context, listen: false);
-    bool available = await speechProvider.initialize();
-    if (available) {
-      await _subscribeToVoiceRecognition(speechProvider);
-      _startListening(speechProvider);
-    } else {
-      _logger.w("The user has denied the use of speech recognition.");
-    }
   }
 
-  Future<void> _subscribeToVoiceRecognition(
-      SpeechToTextProvider speechProvider) async {
-    await FlutterVolumeController.updateShowSystemUI(
-        false); // Hide system volume UI
-    await FlutterVolumeController.setMute(true,
-        stream: AudioStream.alarm); // Set volume to 0 to silence feedback
-
-    _logger.d("Subscribing to voice recognition...");
-
-    _subscription = speechProvider.stream.listen((event) async {
-      if (event.eventType == SpeechRecognitionEventType.finalRecognitionEvent) {
-        var result = event.recognitionResult!;
-        _logger.i("Final result: ${result.recognizedWords}");
-        await _handleSpeechResult(result);
-        _startListening(speechProvider); // Restart listening
-      } else if (event.eventType == SpeechRecognitionEventType.errorEvent) {
-        // _logger.e("Error: ${event.error?.errorMsg}");
-        _startListening(speechProvider); // Restart listening on error
+  Future<void> _initializeSpeechRecognition() async {
+    try {
+      bool available = await speech.initialize(onStatus: listener);
+      if (available) {
+        // TODO: totally disable app sounds -> below code does not mute the recording notifications
+        // await FlutterVolumeController.updateShowSystemUI(false); // Hide system volume UI
+        // await FlutterVolumeController.setMute(true, stream: AudioStream.alarm); // Set volume to 0 to silence feedback
+        _logger.d("Initializing voice recognition...");
+        // listenFor and pauseFor are strictly equal to 5 due to android system voice listen duration
+        speech.listen( onResult: _handleSpeechResult, listenFor: const Duration(seconds: 5), pauseFor: const Duration(seconds: 5),listenOptions: speechListenOptions);
       }
-    });
+      else {
+        _logger.w("The user has denied the use of speech recognition.");
+      }
+    }catch (error, stackTrace) {
+      _logger.e(error.toString(), stackTrace: stackTrace);
+    }
   }
 
-  void _startListening(SpeechToTextProvider speechProvider) {
-    speechProvider.listen(
-      listenFor: const Duration(seconds: 30),
-      pauseFor: const Duration(seconds: 10),
-      partialResults: false,
-      onDevice: false,
-      listenMode: ListenMode.confirmation,
-    );
+  void _restartListening() async {
+    await speech.stop();
+    // listenFor and pauseFor are strictly equal to 5 due to android system voice listen duration
+    speech.listen( onResult: _handleSpeechResult, listenFor: const Duration(seconds: 5), pauseFor: const Duration(seconds: 5),listenOptions: speechListenOptions);
```

Comment:
maybe we should do the pauseFor duration to br something shorter?

Reply:
This option is malfunctioning in speech_to_text package - android does not allow to listen longer than 5 secs, and for some reason if the value of pauseFor or listenFor is not exactly that value (5 sec) the whole listening process is shorter (~3 secs.) which puts more load on the whole system. Originally those values were about 30 secs, which as i wrote is not possible in android.